src/main/java/teammates/ui/controller/StudentFeedbackResultsPageData.java

```diff
@@ -1,13 +1,225 @@
 package teammates.ui.controller;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
 import teammates.common.datatransfer.AccountAttributes;
-import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.datatransfer.FeedbackParticipantType;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackQuestionDetails;
+import teammates.common.datatransfer.FeedbackResponseAttributes;
+import teammates.common.datatransfer.FeedbackResponseCommentAttributes;
 import teammates.common.datatransfer.FeedbackSessionResultsBundle;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Url;
+import teammates.ui.template.FeedbackResultsQuestionDetails;
+import teammates.ui.template.FeedbackResultsResponse;
+import teammates.ui.template.FeedbackResultsResponseTable;
+import teammates.ui.template.StudentFeedbackResultsQuestionWithResponses;
+import teammates.ui.template.StudentFeedbackResultsResponseComment;
 
 public class StudentFeedbackResultsPageData extends PageData {
     public FeedbackSessionResultsBundle bundle = null;
+    private String registerMessage;
+    private List<StudentFeedbackResultsQuestionWithResponses> feedbackResultsQuestionsWithResponses;
     
     public StudentFeedbackResultsPageData(AccountAttributes account, StudentAttributes student) {
         super(account, student);
     }
+    
+    public void init(String regKey, String email, String courseId, 
+                       Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionsWithResponses) {
+        
+        String joinUrl = new Url(Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
+                                                   .withRegistrationKey(regKey)
+                                                   .withStudentEmail(email)
```

Comment:
Is `email` any different from `student.email`?
`student.email` is used in the *PageAction, so it may be referring to the same String.